setup.py

```diff
@@ -73,12 +74,13 @@
         'pyproj',
         'Py6S>=1.7.0',
         'shapely',
-        'gippy @ https://github.com/Applied-GeoSolutions/gippy/archive/v0.3.11.tar.gz#egg=gippy-0.3.11',
+        'gippy',
         'homura==0.1.3',
         'python-dateutil',
         'pydap==3.2',
         'pysolar==0.6',
         'dbfread==2.0.7',
+        # TODO this format doesn't seem to work and I can find no documentation for it:
```

Comment:
since pip 19.1 --> https://pip.pypa.io/en/stable/reference/pip_install/#id29

Reply:
Yeah, silly me, I thought I could find the relevant docs in setuptools, _since we're making a setuptools call here_, but as I later learned, it doesn't seem to support these "pep 508 urls," even though pip does.

I later learned that ubuntu 18.04's python3-pip package installs pip3 circa 9.0.  Since it's a system installer we can't presume that version of pip _isn't_ installed, which means we can't install our own system pip.  We can't upgrade the system pip either, because `pip3 install -U pip` or whatever doesn't play nice with the pip3 installed by the OS package (there are warnings against that in the docs which I later discovered).

So it looks like we can't use those @-url style dependencies because we have to play nice with setuptools and/or old pip3 installed on ubuntu.

Meanwhile we shouldn't use `dependency_links` either because it's been removed from new pip3/setuptools and we have to support that too.

I would love to learn that I'm wrong.  Installing some dependencies packages separately like that is definitely an antipattern.

Reply:
Supporting both seems like a pain.  I haven't run into a problem with hackishly updating pip over the system installed one, but that still seems bad.  An alternative is to uninstall it and replace it with pip from [PyPA - get-pip.py](https://bootstrap.pypa.io/get-pip.py).

Reply:
I'm not sure it's safe to uninstall an ubuntu package.  These packages seem to depend on `python3-pip` for instance:

```
root@c6f92820de9d:/# apt-cache rdepends python3-pip
python3-pip
Reverse Depends:
  python3.6-venv
  python3.7-venv
  python3.6-venv
  thonny
  python3.7-venv
  apertium-apy
  python3-pypandoc
  python3-jupyter-core
  dhcpcanon
```

The best we can do is probably have gips refuse to install if pip3 18 isn't installed, and if there's no pip3, install from `get-pip.py`.